packages/shared/src/logger/logger.ts

```diff
@@ -91,4 +104,15 @@ class Logger {
   }
 }
 
-export { Logger };
+let logger: Logger;
+
+// This should be called with the name once at the entry point.
+// For example, with the Playwright plugin, it is called in the Reporter interface constructor.
+function initLogger(name: string) {
```

Comment:
We should probably warn if it's called more than once?

Reply:
I would prefer to error loudly rather than warn. Otherwise developers will likely not realize it when they introduce multiple calls to `initLogger`, and logging is a critical piece of our infrastructure for customer support.

Thoughts? @bvaughn

Reply:
I chose to use a warning because if we accidentally ship two `inits`, I don’t want to break userland functionality. For example, if the second `init` is in a code path a developer hasn’t tested, we wouldn’t see an error or warning at all. It would break in userland.

Reply:
It's fine to keep it as-is for now, in my opinion, until #562.